### DISCUSSION TITLE
Civil 3d Pull - Failing to connect

### DISCUSSION THREAD

**Discussion by alex-d-richards** [BHoM/BHoM#1415] [Q&A]
Hi, 

I'm trying to pull data from Civils 3d 2021 but get a  failed to connect error on the pull component. 

Do I need to activate the adapter from within Civil 3d like you would with Revit? I don't see a BHoM toolbar or option from within C3D for this?

I see that Civil 3d adapter allows you to configure the import and outport but again not sure how this can be adjusted from Civil 3d. 

Hopefully someone can point me in the right direction. 

UPDATE

Looking back at a very early Issue it looks although a socket/ adapter active is required similar to Revit. 

[https://github.com/BHoM/Civil3D_Toolkit/issues/2]

I'm therefore pretty sure I'm missing the .addin for C3D which is possible due to how I have built/ installed BHoM. 

I had originally tried to build from source following:  [https://github.com/BHoM/documentation/wiki/Getting-started-for-developers](url) but didn't manage to successfully build the project with a number of components missing from the GH UI. 

I therefore used the installer and then built the C3D Toolkit which works fine in GH. 

However I do not generate a .addin for civil 3d? and I'm not sure where this should come from?

Hopefully, on the right track but would be great if someone who knows allot more about this toolkit @FraserGreenroyd can point me in the correct direction 

Grasshopper Definition

![image](https://user-images.githubusercontent.com/69122437/179207560-b6895c79-3732-4269-9158-44e9aa8244cb.png)

Civil 3d Tool Bar

![image](https://user-images.githubusercontent.com/69122437/179207695-13a2bcec-3507-4b94-959e-0df91329a371.png)

Many Thanks
Alex

**Comment by FraserGreenroyd** [comment]
Hi @alex-d-richards - I'm not sure if the Civil3D Toolkit has been built to support C3D 2021,  I think @cesarecaoduro was investigating some C3D enhancements though, so might be worth reaching out with them to see how far they got?
